libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/routeprogress/RouteLegProgressTest.java

```diff
@@ -117,174 +77,129 @@ public void previousStep_returnsPreviousStep() {
   }
 
   @Test
-  public void stepIndex_returnsCurrentStepIndex() {
-    RouteProgress routeProgress = RouteProgress.builder()
-      .stepDistanceRemaining(route.legs().get(0).steps().get(0).distance())
-      .legDistanceRemaining(route.legs().get(0).distance())
-      .distanceRemaining(route.distance())
-      .directionsRoute(route)
-      .stepIndex(3)
-      .legIndex(0)
-      .build();
+  public void stepIndex_returnsCurrentStepIndex() throws Exception {
+    RouteProgress routeProgress = buildDefaultTestRouteProgress();
+
+    routeProgress = routeProgress.toBuilder().stepIndex(3).build();
 
     assertEquals(3, routeProgress.currentLegProgress().stepIndex(), BaseTest.DELTA);
   }
 
   @Test
-  public void fractionTraveled_equalsZeroAtBeginning() {
-    RouteProgress routeProgress = RouteProgress.builder()
-      .stepDistanceRemaining(route.legs().get(0).steps().get(0).distance())
-      .legDistanceRemaining(route.legs().get(0).distance())
-      .distanceRemaining(route.distance())
-      .directionsRoute(route)
-      .stepIndex(0)
-      .legIndex(0)
-      .build();
+  public void fractionTraveled_equalsZeroAtBeginning() throws Exception {
+    RouteProgress routeProgress = buildBeginningOfLegRouteProgress();
 
     assertEquals(0.0, routeProgress.currentLegProgress().fractionTraveled(), BaseTest.DELTA);
   }
 
   @Test
-  public void fractionTraveled_equalsCorrectValueAtIntervals() {
+  public void fractionTraveled_equalsCorrectValueAtIntervals() throws Exception {
+    RouteProgress routeProgress = buildDefaultTestRouteProgress();
+    DirectionsRoute route = buildTestDirectionsRoute();
+    RouteLeg firstLeg = route.legs().get(0);
     double stepSegments = 5000; // meters
 
     // Chop the line in small pieces
```

Comment:
What about extracting this block of code into a private method and give it a name based on the comment? This way the test will be easier to read and understand and comment will become superfluous so it won't be necessary. Same thing with `double stepSegments = 5000; // meters` ☝️